### DISCUSSION TITLE
Don't render/show a dataframe index?

### DISCUSSION THREAD

**Discussion by astrowonk** [quarto-dev/quarto-cli#1522] [Q&A] [answered]
Using pandas/python as my language, if a code block returns a dataframe, the dataframe renders fairly nicely but with the index in bold. My indices are usually just sequential or sometimes random (thanks to sorting) numbers. 

I guess as a workaround I could set the index to one of my columns (though then it'd become bold text), but is there a way to have quarto just not render the index at all?

**Comment by astrowonk** [comment]
> Using pandas/python as my language, if a code block returns a dataframe, the dataframe renders fairly nicely but with the index in bold. My indices are usually just sequential or sometimes random (thanks to sorting) numbers.
> 
> I guess as a workaround I could set the index to one of my columns (though then it'd become bold text), but is there a way to have quarto just not render the index at all?

So, I tried to get around this by just setting index to one of the columns, but now the headers look like this:

<img width="618" alt="Screen Shot 2022-08-13 at 7 44 51 AM" src="https://user-images.githubusercontent.com/13702392/184492697-a925c518-432d-4c02-bb73-2efc6f4dc2a5.png">

which was unexpected. (This is using python, it is a pandas dataframe where I just set the index so it only has one column)

**Comment by astrowonk** [comment, marked as answer]
The best workaround for now this this:

```
from IPython.display import display, HTML
display(HTML(df.to_html(index=False)))

and a simple wrapper function makes this easier
def format_df(df):
    """format the dataframe for display without index"""
    return display(HTML(df.to_html(index=False)))

```

**Comment by nathancarter** [comment]
This workaround only applies when rendering HTML.  I attempted a similar workaround for Latex, but it fails because I have `$` in some of my table columns.  An official solution for this would be better.

**Reply by nathancarter** [reply]
Turns out this is a built-in feature of pandas:

```python
df.style.hide(axis="index")
```
